lte/gateway/c/core/oai/tasks/amf/amf_recv.cpp

```diff
@@ -149,39 +237,83 @@ int amf_handle_service_request(
-              if (smf_context->pdu_address.pdn_type == IPv4) {
-                inet_ntop(
-                    AF_INET, &(smf_context->pdu_address.ipv4_address.s_addr),
-                    ip_str, INET_ADDRSTRLEN);
-              }
-
-              OAILOG_DEBUG(
-                  LOG_NAS_AMF,
-                  "Sending session request to SMF on service request for "
-                  "sessiond %u\n",
-                  session_id);
-              notify_ue_event_type = UE_SERVICE_REQUEST_ON_PAGING;
-              // construct the proto structure and send message to SMF
-              amf_smf_notification_send(
-                  ue_id, ue_context, notify_ue_event_type);
+        OAILOG_DEBUG(LOG_NAS_AMF, "Service request type is %s \n",
+        (msg->service_type.service_type_value == SERVICE_TYPE_DATA)?
+         "Data" : "Mobile Terminated Services");
+
+        if (REGISTERED_CONNECTED == ue_context->mm_state)
+        {
+            amf_sap.primitive                     = AMFAS_ESTABLISH_CNF;
+            amf_sap.u.amf_as.u.establish.ue_id    = ue_id;
+            amf_sap.u.amf_as.u.establish.nas_info = AMF_AS_NAS_INFO_SR;
+
+            for (uint16_t session_id = 1; session_id < (sizeof(session_id) * 8);
+                    session_id++) {
+                if (msg->uplink_data_status.uplinkDataStatus & (1 << session_id)) {
+                    std::shared_ptr<smf_context_t> smf_context =
+                        amf_get_smf_context_by_pdu_session_id(ue_context, session_id);
+		    if (smf_context && (smf_context->pdu_session_state == ACTIVE)){
+                        pdu_session_status |= (1 << session_id);
+                    }
+                }
+            }
+
+            amf_sap.u.amf_as.u.establish.pdu_session_status_ie = AMF_AS_PDU_SESSION_STATUS;
+            amf_sap.u.amf_as.u.establish.pdu_session_status = pdu_session_status;
+            amf_sap.u.amf_as.u.establish.guti = ue_context->amf_context.m5_guti;
+            rc                                = amf_sap_send(&amf_sap);
+        }
+	else
+        {
+            for (uint16_t session_id = 1; session_id < (sizeof(session_id) * 8);
+                    session_id++) {
+                if (msg->uplink_data_status.uplinkDataStatus & (1 << session_id)) {
+                    std::shared_ptr<smf_context_t> smf_context =
+                        amf_get_smf_context_by_pdu_session_id(ue_context, session_id);
+                    if (smf_context) {
+                        pdu_session_status |= (1 << session_id);
+                    }
+                }
+            }
+
+            /* Start Service response timer */
+            ue_context->pending_service_response = true;
+            ue_context->pdu_session_status = pdu_session_status;
+            ue_context->pdu_session_reactivation_status = pdu_session_status;
+            ue_context->service_response_timer.id = amf_app_start_timer(
+                    SERVICE_RESEPONE_TIMER_EXPIRY_MSECS,
```

Comment:
Timer in UE T3517 can be helpful to figure out if service request needs to be aborted.

Reply:
Taken care